runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -391,12 +388,22 @@ function TSHighlighter._on_win(_, _win, buf, topline, botline)
   if not self then
     return false
   end
-  self.tree:parse({ topline, botline + 1 })
-  self:prepare_highlight_states(topline, botline + 1)
-  self.redraw_count = self.redraw_count + 1
+  local range = { topline, botline + 1 }
+  self.tree:parse(range, function(_, timeout_reached)
+    if not timeout_reached then
+      self:_async_parse_callback(range)
+    end
+  end)
   return true
 end
 
+--- @param range [integer, integer]
+function TSHighlighter:_async_parse_callback(range)
+  self:prepare_highlight_states(unpack(range))
+  self.redraw_count = self.redraw_count + 1
```

Comment:
This is a private variable not used anywhere that likely just exists for debugging purposes? Moving it without renaming doesn't seem to make much sense. The "redraw count" increments in the `on_win` callback. I would suggest any of; leaving `redraw_count` in `on_win`, adding `parse_count?` in `_asyn_parse_callback`, removing both?